### PR TITLE
Improve cache auto-configuration for Redis

### PR DESCRIPTION
**Pull request by ryonday**
Expose key prefix, TTL and null value settings for spring-data-redis' RedisCacheConfiguration in Spring .properties/yml configuration files.

Example:

```
spring.cache.redis.ttl=PT15M
spring.cache.redis.keyPrefix=foo
spring.cache.redis.useKeyPrefix=false
spring.cache.redis.cacheNullValues=false
```

Fixes gs-10795

### CODE REVIEW

**Comment by ryonday** [comment]
Addresses [this](https://github.com/spring-projects/spring-boot/issues/10795)

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java [hunk 1]
```diff
@@ -233,4 +242,72 @@ public void setConfig(Resource config) {
...
+	/**
+	 * Redis-specific cache properties. Properties set will be used as the defaults for
+	 * all Redis caches.
+	 *
+	 * @see org.springframework.data.redis.cache.RedisCacheConfiguration
+	 */
+	public static class Redis {
+
+		/**
+		 * The TTL for entries. By default, entries do not expire, a value of zero
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java]: Configuration key descriptions do not start with `The`, `A`, etc. Please look at [existing keys](https://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html) for inspiration

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java [hunk 2]
```diff
@@ -233,4 +242,72 @@ public void setConfig(Resource config) {
...
+		 * indicates an eternal entry, and the given value will be converted to seconds..
+		 *
+		 * @see Duration#parse(CharSequence)
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java]: We don't make such `@see` link here. We use the javadoc to infer the description. Can you please remove them all?

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java [hunk 3]
```diff
@@ -233,4 +242,72 @@ public void setConfig(Resource config) {
...
+		 */
+		private Duration ttl = Duration.ZERO;
+
+		/**
+		 * Whether to allow caching of {@literal null} values.
+		 */
+		private boolean cacheNullValues = true;
+
+		/**
+		 * The non-null Redis key prefix to use instead of the default one.
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java]: I am not sure what you mean by "non-null" here.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java [hunk 4]
```diff
@@ -233,4 +242,72 @@ public void setConfig(Resource config) {
...
+		public boolean isCacheNullValues() {
+			return this.cacheNullValues;
+		}
+
+		public Redis setCacheNullValues(boolean cacheNullValues) {
+			this.cacheNullValues = cacheNullValues;
+			return this;
+		}
+
+		public Optional<String> getKeyPrefix() {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheProperties.java]: Configuration properties must be JavaBean properties (the type must match) and we don't support `Optional` here.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/RedisCacheConfiguration.java [hunk 5]
```diff
@@ -57,13 +59,38 @@
 
 	@Bean
 	public RedisCacheManager cacheManager(RedisConnectionFactory redisConnectionFactory) {
+
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/RedisCacheConfiguration.java]: Please refrain from making unnecessary formatting changes.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/RedisCacheConfiguration.java [hunk 6]
```diff
@@ -57,13 +59,38 @@
...
 		RedisCacheManagerBuilder builder = RedisCacheManager
-				.builder(redisConnectionFactory);
+				.builder(redisConnectionFactory).cacheDefaults(getConfiguration());
+
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/RedisCacheConfiguration.java]: Idem

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/RedisCacheConfiguration.java [hunk 7]
```diff
@@ -57,13 +59,38 @@
...
 		List<String> cacheNames = this.cacheProperties.getCacheNames();
 		if (!cacheNames.isEmpty()) {
 			builder.initialCacheNames(new LinkedHashSet<>(cacheNames));
 		}
+
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/RedisCacheConfiguration.java]: Idem

**Comment by ryonday** [comment]
Thanks so much for the detailed feedback, @snicoll; I hope that the follow-up commit has addressed your concerns! 

I'm happy to be able to give back in some small way to a project that I've used so much throughout my career.

**Comment by snicoll** [comment]
@ryonday Thanks for the feedback. They did partially and I had an extra polish pass. It's now merge in `master`.

**Comment by ryonday** [comment]
Thanks so much for the polish pass, I have a few more small PR ideas and it helps to have a template for the preferred conventions and style to make it easier on reviewers! 

Cheers!
